client/components/site-selector/index.jsx

```diff
@@ -615,9 +566,7 @@ const mapState = ( state ) => {
 
 	return {
 		hasLoadedSites: hasLoadedSites( state ),
-		sites: getSites( state ),
-		showRecentSites: get( user, 'visible_site_count', 0 ) > 11,
-		recentSites: getPreference( state, 'recentSites' ),
+		sites: getSites( state, { shouldSort: false } ),
```

Comment:
This is so we don't sort twice: one pre-sort before the list is rendered, and then sorting right before rendering the items.